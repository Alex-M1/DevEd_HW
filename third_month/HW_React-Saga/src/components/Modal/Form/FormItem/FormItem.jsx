import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { StFormItem } from '../styled';

function FormItem({ title, value, placeholder, onChange, isRef }) {
  const onInputChange = (e) => {
    onChange(e.target.value);
  };
  const inputRef = useRef(null);

  useEffect(() => {
    if (isRef) {
      inputRef.current.focus();
    }
  }, []);

  return (
    <StFormItem>
      <span>{title}</span>
      <input
        ref={inputRef}
        value={value}
        onChange={onInputChange}
        placeholder={placeholder}
      />
    </StFormItem>
  );
}

export default FormItem;

FormItem.propTypes = {
  isRef: PropTypes.bool,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
};
