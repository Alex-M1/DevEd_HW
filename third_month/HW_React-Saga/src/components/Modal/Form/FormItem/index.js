import { connect } from 'react-redux';
import { setInputValue } from '../../../../store/actions';
import { iptValue } from '../../../../store/selectors';
import FormItem from './FormItem';

const mapStateToProps = (state, props) => ({
  value: iptValue(state, props.type),
});

const mapDispatchToProps = (dispatch, props) => ({
  onChange: (value) => dispatch(setInputValue(props.type, value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormItem);
