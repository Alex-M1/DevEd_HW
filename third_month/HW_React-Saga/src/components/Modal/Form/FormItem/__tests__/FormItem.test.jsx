import React from 'react';
import FormItem from '../FormItem';

describe('FormItem', () => {
  const props = {
    title: 'title',
    value: 'value',
    placeholder: 'placeholder',
    isRef: true,
    onChange: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<FormItem {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StFormItem', () => {
    const component = global.shallowSmart(<FormItem {...props} />);
    expect(component.find('styled__StFormItem')).toHaveLength(1);
  });
  it('Should have span', () => {
    const component = global.shallowSmart(<FormItem {...props} />);
    expect(component.find('span')).toHaveLength(1);
  });
  it('Should have input', () => {
    const component = global.shallowSmart(<FormItem {...props} />);
    expect(component.find('input')).toHaveLength(1);
  });
  it('Should be a change', () => {
    const newValue = 'MockValueNew';
    const component = global.mount(<FormItem {...props} />);
    component.find('input').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledWith(newValue);
  });
});
