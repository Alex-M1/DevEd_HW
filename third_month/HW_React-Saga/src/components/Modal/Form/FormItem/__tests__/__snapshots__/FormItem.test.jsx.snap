// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem Should match snapshot 1`] = `"<div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>title</span><input value=\\"value\\" placeholder=\\"placeholder\\"/></div>"`;
